src/main/resources/avro/common.avdl

```diff
@@ -17,31 +29,88 @@ error GAException {
 }
 
 /**
-An abstraction for referring to a genomic position, in relation to some
-already known reference. For now, represents a genomic position as a reference
-name, a base number on that reference (0-based), and a flag to say if it's the
-forward or reverse strand that we're talking about.
+Each base in a sequence graph has two sides, `PLUS` and `MINUS`.
+
+`PLUS` represents forwards, or the direction of increasing coordinates, while
+`MINUS` represents reverse-complement, or the direction of decreasing
+coordinates.
+*/
+enum Side {
+  PLUS,
+  MINUS
+}
+
+/**
+An abstraction for referring to a genomic position on a sequence.
+
+For now, represents a genomic position as a sequence ID, a base offset on that
+sequence (0-based), and a flag to say if it's the forward or reverse strand that
+we're talking about.
 */
-// TODO: Add support here for universally unique base IDs.
 record Position {
   /**
-  The name of the reference (or, more technically, the scaffold) in whatever
-  reference set is being used. Does not generally include a "chr" prefix, so for
-  example "X" would be used for the X chromosome.
+  The ID of the sequence on which the `Position` is located.
   */
-  string referenceName;
+  string id;
```

Comment:
Do we  need this back-reference? We're just embedding these into `Segment` objects, and there's no reason to look them up independently. `Position` should be as lightweight as possible, so I would vote to remove this.

We're running into major problems with backwards compatibility here though. By changing the `Position` type, the old protocol won't work any more. Perhaps we should leave  `Position` alone and use a new `Location` object? We could then slowly deprecate the old reference interface?